src/diagnostics.rs

```diff
@@ -758,6 +767,12 @@ pub struct ResolvedLocation {
     pub range: Range<usize>,
 }
 
+impl ResolvedLocation {
+    pub fn is_internal(&self) -> bool {
+        self.range == (0..0)
```

Comment:
could you check against SourceRange::undefined instead? I know it's _currently_ the same, it just feels clearer.